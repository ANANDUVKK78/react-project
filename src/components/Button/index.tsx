import React, { ReactElement } from 'react'
import {Props} from './types';

const Button = (props: Props): ReactElement => {
    return (
        <button
            type={props.type}
            className={props.className}
            disabled={props.disabled}
            onClick={props.click}
        >
            {props.label}
        </button>
    )
}



export default Button
