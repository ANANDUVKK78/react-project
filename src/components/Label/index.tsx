import React, { ReactElement } from 'react'
import { Props } from './types'


const Label = (props:Props): ReactElement => {
    return <label className={props.className}>{props.text}</label>
}




export default Label
