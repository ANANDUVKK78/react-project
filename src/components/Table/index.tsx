import React, { ReactElement } from 'react'
import { Props } from './types'

const Table = (props: Props): ReactElement => {
    return (
        <>
            <table  className={props.className} >
                {props.children}
            </table>
        </>
    )
}



export default Table
