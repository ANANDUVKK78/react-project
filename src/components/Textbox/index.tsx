import React, { ReactElement} from 'react'
import { Props } from './types';

const Textbox = (props: Props): ReactElement => {
    return (
        <input
            type={props.type}
            name={props.name}
            className={props.className}
            defaultValue={props.value}
            disabled = {props.disabled}
            onBlur={props.blur}
            onChange={props.change}
            placeholder={props.placeholder}
        ></input>
    )
}




export default Textbox
