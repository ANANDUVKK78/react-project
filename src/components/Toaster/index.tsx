import React, { ReactElement} from 'react';
import styles from './styles.scss'

const Toaster = (): ReactElement => {
    return (
       <div className ={styles.toasterContainer}>
           Internal server error
       </div>
    )
}




export default Toaster
