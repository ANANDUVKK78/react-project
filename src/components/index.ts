import Button from "./Button";
import Label from "./Label";
import Table from "./Table";
import Textbox from "./Textbox";
import Toaster from "./Toaster";

export {Button,Label,Table,Textbox,Toaster};
