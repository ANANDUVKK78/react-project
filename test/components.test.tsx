import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import {Button,Textbox,Label,Table} from 'components'

configure({ adapter: new Adapter() })

describe('Testing Components', () => {
    it('testing button component', () => {
        const mockFunc: jest.Mock = jest.fn()

        const wrapper = shallow(
            <Button
                type="button"
                label="Reg"
                className="sample"
                disabled={false}
                click={mockFunc}
            ></Button>
        )
        wrapper.find('button').simulate('click')
        expect(wrapper.find('button')).toHaveLength(1)
        expect(wrapper.find('button').text()).toBe('Reg')
        expect(wrapper.find('button').props().className).toBe('sample')
        expect(wrapper.find('button').props().disabled).toBe(false)
        expect(mockFunc.mock.calls.length).toEqual(1)

        const tree = renderer.create(<Button label="reg" />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('testing textbox component', () => {
        const mockChangeFunc: jest.Mock = jest.fn()
        const mockBlurFunc: jest.Mock = jest.fn()

        const wrapper = shallow(
            <Textbox
                type="text"
                className="sample"
                placeholder="sample"
                disabled={true}
                change={mockChangeFunc}
                blur={mockBlurFunc}
            ></Textbox>
        )
        wrapper.find('input').simulate('change')
        wrapper.find('input').simulate('blur')
        expect(wrapper.find('input')).toHaveLength(1)
        expect(wrapper.find('input').props().type).toBe('text')
        expect(wrapper.find('input').props().className).toBe('sample')
        expect(wrapper.find('input').props().placeholder).toBe('sample')
        expect(wrapper.find('input').props().disabled).toBe(true)
        expect(mockChangeFunc.mock.calls.length).toEqual(1)
        expect(mockBlurFunc.mock.calls.length).toEqual(1)
        const tree = renderer.create(<Textbox />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('testing label component', () => {
        const wrapper = shallow(<Label text="text" className="sample"></Label>)
        expect(wrapper.find('label')).toHaveLength(1)
        expect(wrapper.find('label').text()).toBe('text')
        expect(wrapper.find('label').props().className).toBe('sample')
        const tree = renderer.create(<Label />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('testing Table component', () => {
        const wrapper = shallow(
            <Table className="sample">
                <tr>
                    <td>sample</td>
                </tr>
            </Table>
        )
        expect(wrapper.find('table')).toHaveLength(1)
        expect(wrapper.find('table').props().className).toBe('sample')
        const tree = renderer
            .create(
                <Table>
                    <tr></tr>
                </Table>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
