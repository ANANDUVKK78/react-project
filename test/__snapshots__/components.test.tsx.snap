// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Components testing Table component 1`] = `
<table>
  <tr />
</table>
`;

exports[`Testing Components testing button component 1`] = `
<button>
  reg
</button>
`;

exports[`Testing Components testing label component 1`] = `<label />`;

exports[`Testing Components testing textbox component 1`] = `<input />`;
